### DISCUSSION TITLE
Error when converting array of strings to XML

### DISCUSSION THREAD

**Discussion by kinekt4** [downforcetech/fluidxml.php#48] [Questions]
I'm trying to convert an array of strings to XML.

I get "Invalid Character Error" when calling `addChild()`:

    $arrA = array('abc 123', 'def 456');
    $xml = new FluidXml();
    $xml->addChild($return);

    <exception class="DOMException" message="Invalid Character Error">

When removing the space, the exception is not thrown:

    <?xml version="1.0" encoding="UTF-8"?>
    <doc>
        <abc123/>
        <def456/>
    </doc>

I'm expecting something like this:

    <?xml version="1.0" encoding="UTF-8"?>
    <doc>
        <idx>abc 123</idx>
        <idx>def 456</idx>
    </doc>

I know I can achieve that with the following:

    $xml = new FluidXml();
    $xml->add('idx', 'abc 123');
    $xml->add('idx', 'def 456');

However, I was hoping FluidXml can do it automagically for me.

Is there a solution/workaround I can use?

**Comment by daniele-orlando** [comment]
FluidXml has a well defined protocol on how it handles the array and your expressed case is quite uncommon/exceptional compared to the general pattern of an array of strings. In addition it is not possible to generalize your specific case (the key name that works for you will not work for someone else).

What you should do is mapping your array to the XML structure you want:
```php
$list = [ 'abc 123', 'def 456' ];
$items = array_map(function ($it) { return [ 'idx' => $it ];  }, $list);
$xml->addChild($items);
```

**Comment by kinekt4** [comment]
Thanks for your response. It has been useful.
